colossalai/shardformer/policies/t5.py

```diff
@@ -228,13 +229,9 @@ def distribute_t5_layers(num_encoder_layers: int, num_decoder_layers: int,
         def objective(num_encoder_stages):
             return abs(num_encoder_layers / num_encoder_stages - num_decoder_layers / (num_stages - num_encoder_stages))
 
-        num_encoder_stages = 0
-        optimal_diff = 2**31 - 1
-        for i in range(1, num_stages):
-            attempt = objective(i)
-            if attempt < optimal_diff:
-                num_encoder_stages = i
-                optimal_diff = attempt
+        diff = range(1, num_stages)
+        num_encoder_stages = np.argmin([objective(i) for i in diff]) + 1
```

Comment:
`diff` is not meaningful enough. You can just move it in for-loop.

Reply:
ok! , I put this in the for loop